### PR TITLE
add OSMnx usage example

### PR DESCRIPTION
**Pull request by gboeing**
@jarrodmillman I've added a simple OSMnx usage example for your review.

Cross-reference #4366 #4298.

### CODE REVIEW

**Comment by jarrodmillman** [comment]
LGTM!  It would be great if @ljwolf, @martinfleis could take a quick look.

@gboeing maybe you could take a quick look at https://github.com/ljwolf/networkx/pull/1.

File: examples/geospatial/plot_osmnx.py [hunk 1]
```diff
@@ -3,15 +3,47 @@
...
+https://osmnx.readthedocs.io/ for OSMnx documentation and usage.
 """
 
-import osmnx as ox
 import networkx as nx
+import osmnx as ox
+
+ox.config(use_cache=True, log_console=True)
+
+# download street network data from OSM and construct a MultiDiGraph model
```
Comment (dschult) [inline examples/geospatial/plot_osmnx.py:21]: Is this comment correct?

**Review by martinfleis — APPROVED** [review]

**Review by ljwolf — APPROVED** [review]

File: examples/geospatial/plot_osmnx.py (hunk 1 above)
Comment (jarrodmillman) [reply examples/geospatial/plot_osmnx.py:21]: I believe so.  I didn't look at the implementation, but if I turn off my wireless network it produces the following error
```
In [1]:  import osmnx as ox

In [2]:  G = ox.graph_from_point((37.79, -122.41), dist=750, network_type="drive")
...
ConnectionError: HTTPConnectionPool(host='overpass-api.de', port=80): Max retries exceeded with url: /api/status (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7e6de14c10>: Failed to establish a new connection: [Errno -2] Name or service not known'))
...
```
and if I reenable it I get
```
In [3]:  G = ox.graph_from_point((37.79, -122.41), dist=750, network_type="drive")

In [4]: G
Out[4]: <networkx.classes.multidigraph.MultiDiGraph at 0x7f7e6c8286a0>
```

**Review by jarrodmillman — APPROVED** [review]

File: examples/geospatial/plot_osmnx.py (hunk 1 above)
Comment (martinfleis) [reply examples/geospatial/plot_osmnx.py:21]: Yes, it is correct.

**Review by dschult — APPROVED** [review]

**Pull request merged**